var/spack/repos/builtin/packages/lulesh/package.py

```diff
@@ -23,33 +23,66 @@
 # Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA 02111-1307 USA
 ##############################################################################
 from spack import *
-import os
 
 
-class Lulesh(Package):
-    """Livermore Unstructured Lagrangian Explicit Shock Hydrodynamics (LULESH)
+class Lulesh(MakefilePackage):
+    """LULESH is a highly simplified application, hard-coded to only
+    style typical in scientific C or C++ based applications. Hard
+    code to only solve a Sedov blast problem with analytic answer
     """
-
+    tags = ['proxy-app']
     homepage = "https://codesign.llnl.gov/lulesh.php"
     url      = "https://codesign.llnl.gov/lulesh/lulesh2.0.3.tgz"
 
-    version("2.0.3", "336644a8750f71c7c6b9d2960976e7aa")
+    version('2.0.3', '336644a8750f71c7c6b9d2960976e7aa')
 
-    patch("remove_defaults.patch")
+    variant('mpi', default=True, description='Build with MPI support')
+    variant('openmp', default=True, description='Build with OpenMP support')
+    variant('visual', default=False,
+        description='Build with Visualization support (Silo, hdf5)')
 
-    variant('mpip', default=False)
+    depends_on('mpi', when='+mpi')
+    depends_on('silo', when='+visual')
+    depends_on('hdf5', when='+visual')
 
-    depends_on("mpi", type="build")
-    depends_on("mpip", when="+mpip")
+    @property
+    def build_targets(self):
+        targets = []
+        cxxflag = ' -g -O3 -I. '
+        ldflags = ' -g -O3 '
 
-    def install(self, spec, prefix):
-        if '+mpip' in spec:
-            os.environ["LDFLAGS"] = " -lmpiP -ldwarf -lelf"
+        targets.append('CXX = {0} {1}'.format(spack_cxx, ' -DUSE_MPI=0 '))
+        if '+mpi' in self.spec:
+            targets.append(
+                'MPI_INC = {0}'.format(self.spec['mpi'].prefix.include))
+            targets.append('MPI_LIB = {0}'.format(self.spec['mpi'].prefix.lib))
+        if '+visual' in self.spec:
+            targets.append(
+                'SILO_INCDIR = {0}'.format(self.spec['silo'].prefix.include))
+            targets.append(
+                'SILO_LIBDIR = {0}'.format(self.spec['silo'].prefix.lib))
+            cxxflag = ' -g -DVIZ_MESH -I${SILO_INCDIR} '
+            if '+mpi' not in self.spec and '+openmp' not in self.spec:
+                cxxflag += ' -Wall -Wno-pragmas '
+            ldflags = ' -g -L${SILO_LIBDIR} -Wl,-rpath -Wl,'
+            ldflags += '${SILO_LIBDIR} -lsiloh5 -lhdf5 '
+
+        if '+openmp' in self.spec:
+            cxxflag += self.compiler.openmp_flag
+            ldflags += self.compiler.openmp_flag
 
-            if os.uname()[4] == "x86_64":
-                os.environ["LDFLAGS"] += " -lunwind"
+        if '+mpi' in self.spec:
```

Comment:
Can you add this to the `if '+mpi' in self.spec` if-statement above? No need to check this variant multiple times.